tests/factories/cht/generate.js

```diff
@@ -51,7 +60,75 @@ const createData = ({ healthCenter, user, nbrClinics=10, nbrPersons=10 }) => {
   return { clinics, reports, persons };
 };
 
-const createHierarchy = ({ name, user=false, nbrClinics=50, nbrPersons=10 }) => {
+const createClinic = (index, healthCenter) => {
+  const firstName = FIRST_NAMES[index % FIRST_NAMES.length];
+  const lastName = LAST_NAMES[index % LAST_NAMES.length];
+  const personName = `${firstName} ${lastName}`;
+  const personPhoneNumber = PHONE_NUMBERS[index % PHONE_NUMBERS.length];
+
+  const primaryContact = personFactory.build({
+    name: personName,
+    phone: personPhoneNumber
+  });
+
+  const clinic = placeFactory.place().build({
+    type: 'clinic',
+    parent: { _id: healthCenter._id, parent: healthCenter.parent },
+    name: `${personName} Family`,
+    contact: primaryContact
+  });
+
+  primaryContact.parent = { _id: clinic._id, parent: clinic.parent };
+
+  return { clinic, primaryContact };
+};
+
+const createAdditionalPersons = (nbrPersons, clinic) => {
+  return Array.from({ length: nbrPersons - 1 }).map((_, i) => {
+    const additionalPersonName = `${FIRST_NAMES[i % FIRST_NAMES.length]} ${LAST_NAMES[i % LAST_NAMES.length]}`;
+    const additionalPhoneNumber = PHONE_NUMBERS[i % PHONE_NUMBERS.length];
+    return personFactory.build({
+      parent: { _id: clinic._id, parent: clinic.parent },
+      name: additionalPersonName,
+      patient_id: PATIENT_IDS[i % PATIENT_IDS.length],
+      phone: additionalPhoneNumber
+    });
+  });
+};
+
+const createReportsForPerson = (person, user) => {
+  return [
+    deliveryFactory.build(getReportContext(person, user)),
+    pregnancyFactory.build(getReportContext(person, user)),
+    pregnancyVisitFactory.build(getReportContext(person, user))
+  ];
+};
+
+const createDataWithRealNames = ({ healthCenter, user, nbrClinics = 10, nbrPersons = 10 }) => {
+  const clinicsData = Array.from({ length: nbrClinics }).map((_, index) => {
+    const { clinic, primaryContact } = createClinic(index, healthCenter);
+
+    const additionalPersons = createAdditionalPersons(nbrPersons, clinic);
+
+    const allPersons = [primaryContact, ...additionalPersons];
+
+    return { clinic, persons: allPersons };
+  });
+
+  const allPersons = clinicsData.flatMap(data => data.persons);
+  const clinicList = clinicsData.map(data => data.clinic);
+
+  const reports = allPersons.flatMap(person => createReportsForPerson(person, user));
+
+  return { clinics: clinicList, reports, persons: allPersons };
+};
+
+const createData = ({ healthCenter, user, nbrClinics, nbrPersons, useRealNames = false }) => {
+  const createDataFunc = useRealNames ? createDataWithRealNames : createDataWithFixedData;
+  return createDataFunc({ healthCenter, user, nbrClinics, nbrPersons });
```

Comment:
Like this no need to cache a function in a variable 
```suggestion
  if (useRealNames) { 
    return createDataWithRealNames({ healthCenter, user, nbrClinics, nbrPersons });
  }
  return createDataWithFixedData({ healthCenter, user, nbrClinics, nbrPersons });
```